tests/system/python/scripts/package/setup

```diff
@@ -15,38 +15,77 @@ PACKAGE_BUILD_VERSION=$1
-    UNAME="armv7l"
-fi
-
-echo "Build version is "${PACKAGE_BUILD_VERSION}
-echo "ID is "${ID}
-echo "uname is "${UNAME}
-
-sudo cp /etc/apt/sources.list /etc/apt/sources.list.bak
-sudo sed -i "/\b\(archives.fledge-iot.org\)\b/d" /etc/apt/sources.list
-
-sudo apt update && sudo apt upgrade -y && sudo apt update
-echo "==================== DONE update, upgrade, update ============================"
-
-wget -q -O - http://archives.fledge-iot.org/KEY.gpg | sudo apt-key add -
-# FIXME: for centos/rhel
-echo "deb http://archives.fledge-iot.org/${PACKAGE_BUILD_VERSION}/${ID}/${UNAME}/ /" | sudo tee -a /etc/apt/sources.list
-sudo apt update
-
-time sudo DEBIAN_FRONTEND=noninteractive apt install -yq fledge
-echo "==================== DONE INSTALLING Fledge =================="
 
-if [ "${FLEDGE_ENVIRONMENT}" != "docker" ]; then     
-    time sudo apt install -y fledge-gui
-    echo "==================== DONE INSTALLING Fledge GUI ======================"
+if [[ ${OS_NAME} == *"Red Hat"* || ${OS_NAME} == *"CentOS"* ]]; then
+    ID="${ID}7"
+    echo "Build version is "${PACKAGE_BUILD_VERSION}
+    echo "ID is "${ID}
+    echo "uname is "${UNAME}
+    
+    sudo cp -f /etc/yum.repos.d/fledge.repo /etc/yum.repos.d/fledge.repo.bak | true
+    sudo yum update -y
+    echo "==================== DONE update, upgrade ============================"
+    
+    # Configure Fledge repo
+    echo -e "[fledge]\n\
+name=fledge Repository\n\
+baseurl=http://archives.fledge-iot.org/${PACKAGE_BUILD_VERSION}/${ID}/${UNAME}/\n\
+enabled=1\n\
+gpgkey=http://archives.fledge-iot.org/RPM-GPG-KEY-fledge\n\
+gpgcheck=1" | sudo tee /etc/yum.repos.d/fledge.repo
+    
+    sudo yum update -y
+    
+    # Install prerequisites
+    if [[ ${ID} = "centos" ]]; then
+        sudo yum install -y centos-release-scl-rh
+        sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+    elif [[ ${ID} = "rhel" ]]; then
+        sudo yum-config-manager --enable 'Red Hat Software Collections RPMs for Red Hat Enterprise Linux 7 Server from RHUI'
+    fi
+    sudo yum update -y
+    
+    time sudo yum install -y fledge
+    echo "==================== DONE INSTALLING Fledge =================="
+    
+    if [ "${FLEDGE_ENVIRONMENT}" != "docker" ]; then     
+        time sudo yum install -y fledge-gui
+        echo "==================== DONE INSTALLING Fledge GUI ======================"
+    fi
+else
+    if [[ ${ID} = "ubuntu" ]]; then
+        VERSION_ID=$(echo "${VERSION_ID//.}")
+        ID="ubuntu${VERSION_ID}";
+    elif [[ ${ID} = "raspbian" ]]; then
+        ID=$(cat /etc/os-release | grep VERSION_CODENAME | cut -f2 -d"=")
+        UNAME="armv7l"
+    fi
+    
+    echo "Build version is "${PACKAGE_BUILD_VERSION}
+    echo "ID is "${ID}
+    echo "uname is "${UNAME}
+    
+    sudo cp /etc/apt/sources.list /etc/apt/sources.list.bak
+    sudo sed -i "/\b\(archives.fledge-iot.org\)\b/d" /etc/apt/sources.list
```

Comment:
We should prefer putting fledge.list or foglamp.list in /etc/apt/sources.list.d/